File: js/P27.js
// Check is a number is a prime
function isPrime(n){
  n = Math.abs(n);
  if(n in [0,1,2,3]) return true;
  var i = 2;
  ceil = Math.sqrt(n);
  while(i <= ceil){
    if(n%i == 0) return false;
    i++;
  }
  return true;
}

function getFunc(a, b){
  return function(n){
    return n * n + a * n + b;
  }
}

function numOfPrimes(a, b){
  n = 0;
  f = getFunc(a, b);
  while(isPrime(f(n))){
    n++;
  }
  return n;
}

// numOfPrimes(1, 41);

function print(s){
  process.stdout.write(s + "\n");
}

function solveP27(){
  var max = 0, A, B;
  for(var a = -999; a < 1000; a++){
    for(var b = -1000; b <= 1000; b++){
      var n = numOfPrimes(a, b);

      // if(n > 39) {
      //   print("n: " + n  + " a: " + a + " b: " + b);
      // }


      if(n > max) {
        max = n;
        A = a;
        B = b;
        print("n: " + n + " max: " + max + " A: " + A + " B: " + B);
      }
    }
  }
  return {a:A, b:B};
}

solveP27();

File: py/P27.py
from utils import *


def getP27Func(a, b):
    def f(n):
        return n * n + a * n + b
    return f


def numOfPrimes(a, b):
    n = 0
    f = getP27Func(a, b)
    while isPrime(f(n)):
        n = n + 1
    return n


def solveP27():
    m = 0
    for a in range(-999, 1000):
        for b in range(-1000, 1001):
            n = numOfPrimes(a, b)
            # if n > 39:
            #     print(f"n: {n}, a: {a}, b: {b}")
            if n > m:
                m = n
                print(f"max: {m}, a: {a}, b: {b}")


solveP27()
